packages/dds/tree/src/simple-tree/api/configuration.ts

```diff
@@ -345,15 +372,55 @@ export function checkUnion(
 		);
 	}
 
+	if (records.length > 1) {
```

Comment:
I think we also need to check `maps.length > 0 && records.length > 0` as well as `arrays.length > 0 && records.length > 0`, like the check below (`maps.length > 0 && arrays.length > 0`). Although, maybe the first of those two is already covered by your new code below?

Reply:
The first case is covered. The second case is not a potential ambiguity because there is no overlap in the kinds of data they can be constructed from (records only allow record-like input, and arrays only accept iterable input).

The reasons maps and arrays are potentially ambiguous is because both allow iterable input.